src/plugins/data_views/public/services/has_data.ts

```diff
@@ -0,0 +1,144 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart, HttpStart } from 'kibana/public';
+import { DEFAULT_ASSETS_TO_IGNORE } from '../../common';
+import { HasDataViewsResponse, IndicesResponse, IndicesResponseModified } from '../';
+
+export class HasData {
+  private removeAliases = (source: IndicesResponseModified): boolean => !source.item.indices;
+
+  private isUserDataIndex = (source: IndicesResponseModified): boolean => {
+    // filter out indices that start with `.`
+    if (source.name.startsWith('.')) return false;
+
+    // filter out empty sources created by apm server
+    if (source.name.startsWith('apm-')) return false;
+
+    // filter out sources from DEFAULT_ASSETS_TO_IGNORE
+    if (source.name === DEFAULT_ASSETS_TO_IGNORE.LOGS_DATA_STREAM_TO_IGNORE) return false;
+    if (source.name === DEFAULT_ASSETS_TO_IGNORE.METRICS_DATA_STREAM_TO_IGNORE) return false;
+    if (source.name === DEFAULT_ASSETS_TO_IGNORE.METRICS_ENDPOINT_INDEX_TO_IGNORE) return false;
+    if (source.name === DEFAULT_ASSETS_TO_IGNORE.ENT_SEARCH_LOGS_DATA_STREAM_TO_IGNORE)
+      return false;
+
+    return true;
+  };
+
+  start(core: CoreStart) {
+    const { http } = core;
+    return {
+      /**
+       * Check to see if ES data exists
+       */
+      hasESData: async (): Promise<boolean> => {
+        const hasLocalESData = await this.checkLocalESData(http);
+        if (!hasLocalESData) {
+          const hasRemoteESData = await this.checkRemoteESData(http);
+          return hasRemoteESData;
+        }
+        return hasLocalESData;
+      },
+      /**
+       * Check to see if any data view exists
+       */
+      hasDataView: async (): Promise<boolean> => {
+        const dataViewsCheck = await this.findDataViews(http);
+        return dataViewsCheck;
+      },
+      /**
+       * Check to see if user created data views exist
+       */
+      hasUserDataView: async (): Promise<boolean> => {
+        const userDataViewsCheck = await this.findUserDataViews(http);
+        return userDataViewsCheck;
+      },
+    };
+  }
+
+  // ES Data
+
+  private responseToItemArray = (response: IndicesResponse): IndicesResponseModified[] => {
+    const { indices = [], aliases = [] } = response;
+    const source: IndicesResponseModified[] = [];
+
+    [...indices, ...aliases, ...(response.data_streams || [])].forEach((item) => {
+      source.push({
+        name: item.name,
+        item,
+      });
+    });
+
+    return source;
+  };
+
+  private getIndices = async ({
+    http,
+    pattern,
+    showAllIndices,
+  }: {
+    http: HttpStart;
+    pattern: string;
+    showAllIndices: boolean;
+  }): Promise<IndicesResponseModified[]> =>
+    http
+      .get<IndicesResponse>(`/internal/index-pattern-management/resolve_index/${pattern}`, {
```

Comment:
This creates a dependency from data views on data-view management. This endpoint should be moved into the data view service but this can be done as a follow up.

Reply:
I wonder if we could avoid using strings in circumstances like this. Just trying to think of how we can restore some of the type protection we lose when using REST calls. Ideally this path would be exported from the plugin that provides the endpoint.

Reply:
I will create a follow up ticket to remove the getIndices logic from data-view-management and use the hasData service. Which will also move the server route to data-views.

Reply:
Issue link - https://github.com/elastic/kibana/issues/127975